pkg/scheduler/predicates/ha.go

```diff
@@ -134,9 +134,16 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 		}
 
 		podsCount := len(podNames)
-		if podsCount+1 >= int(replicas+1)/2 {
-			glog.Infof("node %s podsCount+1 is %d, int(replicas+1)/2 is %d, skipping",
-				nodeName, podsCount+1, int(replicas+1)/2)
+		// When replicas equals 3, pods on each node should not be greater than 1.
+		maxPodsPerNode := 1
+		if replicas > 3 {
+			// When replicas is greater than 3, we allow more than one pods on one node.
+			maxPodsPerNode = int((replicas + 1) / 2)
```

Comment:
if `replicas =5` , the `maxPodsPerNode` would be `3`.  Is it right?

Reply:
it seems not right... `2, 3` will be possible when there are two nodes.

`int((replicas + 1) / 2)` should be `ceil(replicas / 3)`

replicas | maxPodsPerNode| in three nodes|
--------|--------|--------|
< 3 | not limited, because HA is not possible |
= 3         | 1 | 1, 1, 1| 
4         | 2 | 1, 1, 2|
5         | 2 | 1, 2, 2 |
6         | 2 | 2, 2, 2 |
7        | 3 |  2, 2, 3 |
8        | 3 |  2, 3, 3 |
8        | 3 |  3, 3, 3 |

what do you think? @weekface

Reply:
In the table above, the `maxPodsPerNode` is `2` when `replicas` is `4`, this is not HA.

Reply:
I think the `maxPodsPerNode` is right when `replicas` is `4`, because in three nodes, `1, 1, 2` is ok, but when there are two instances in two nodes (`1, 1`), `maxPodsPerNode` must be `1` at this moment to force the scheduler to find a node which has no TiKV pod. 

I guess we need to update the HA algorithm to take more information into account.

Reply:
`maxPodsPerNode` is always `1` when scheduled TiKV pods less than `3`?

Reply:
The current algorithm is ok for `pd`, but not `tikv`. We need a new HA algorithm for `tikv`.